pkg/utils/v1alpha1/maya.go

```diff
@@ -35,10 +37,17 @@ const (
 
 // ProvisionVolume creates a CstorVolumeClaim(cvc) CR,
 // watcher for cvc is present in maya-apiserver
-func ProvisionVolume(size int64, volName, configclass string) error {
+func ProvisionVolume(
+	size int64,
+	volName,
+	replicaCount,
+	configclass,
+	cspcName string,
+) error {
 
 	annotations := map[string]string{
 		OpenebsConfigClass: configclass,
+		OpenebsCSPCName:    cspcName,
```

Comment:
how about moving to label, becz we need to list CVC based on cspc name? cc: @AmitKumarDas

Reply:
IMO labels is something that once applied should never change in future. If its changed then its a new CVC.
Label is something that truly identifies the resource which needs to be labelled.

We can list or filter using our custom code if K8s does not provide an ability. There are few external projects that have built custom code to list via annotations.

On the whole, if we strongly feel that cspc should be a part of CVC but should not be inside the spec of CVC then it can be in the CVC labels.